iceoryx_hoofs/include/iceoryx_hoofs/log/logstream.hpp

```diff
@@ -14,158 +14,115 @@
+
+    template <typename = std::enable_if_t<std::is_arithmetic<T>::value>>
+    inline explicit constexpr LogOct(const T value) noexcept;
+
+  private:
+    T m_value;
 };
 
-class Logger;
+template <typename T, typename = std::enable_if_t<std::is_arithmetic<T>::value>>
+inline constexpr LogOct<T> oct(const T value) noexcept;
 
 class LogStream
 {
   public:
-    LogStream(Logger& logger, LogLevel logLevel = LogLevel::kWarn) noexcept;
-
-    virtual ~LogStream() noexcept;
-
-    void Flush() noexcept;
-
-    LogStream& operator<<(const char* cstr) noexcept;
-
-    LogStream& operator<<(const std::string& str) noexcept;
-
-    template <typename T, typename std::enable_if<std::is_arithmetic<T>::value, int>::type = 0>
-    LogStream& operator<<(const T val) noexcept
-    {
-        m_logEntry.message.append(cxx::convert::toString(val));
-        m_flushed = false;
-        return *this;
-    }
-
-    template <typename T, typename std::enable_if<std::is_base_of<LogHex, T>::value, int>::type = 0>
-    LogStream& operator<<(const T val) noexcept
-    {
-        std::stringstream ss;
-        // the '+val' is there to not interpret the uint8_t as char and print the character instead of the hex value
-        ss << "0x" << std::hex << +val.value;
-        m_logEntry.message.append(ss.str());
-        m_flushed = false;
-        return *this;
-    }
-
-    template <typename T, typename std::enable_if<std::is_base_of<LogBin, T>::value, int>::type = 0>
-    LogStream& operator<<(const T val) noexcept
-    {
-        m_logEntry.message.append("0b");
-        m_logEntry.message.append(std::bitset<std::numeric_limits<decltype(val.value)>::digits>(val.value).to_string());
-        m_flushed = false;
-        return *this;
-    }
-
-    LogStream& operator<<(const LogRawBuffer& value) noexcept;
+    inline LogStream(const char* file, const int line, const char* function, LogLevel logLevel) noexcept;
+
+    inline virtual ~LogStream() noexcept;
+
+    LogStream(const LogStream&) = delete;
+    LogStream(LogStream&&) = delete;
+
+    LogStream& operator=(const LogStream&) = delete;
+    LogStream& operator=(LogStream&&) = delete;
+
+    inline void flush() noexcept;
+
+    inline LogStream& self() noexcept;
+
+    inline LogStream& operator<<(const char* cstr) noexcept;
+
+    /// @todo iox-#1345 instead of using std::string we could also accept everything with a c_str() method
+    /// and avoid the std::string dependency
+    inline LogStream& operator<<(const std::string& str) noexcept;
+
+    template <typename T, typename std::enable_if_t<std::is_signed<T>::value, int> = 0>
+    inline LogStream& operator<<(const T val) noexcept;
+
+    template <typename T, typename std::enable_if_t<std::is_unsigned<T>::value, int> = 0>
+    inline LogStream& operator<<(const T val) noexcept;
+
+    template <typename T, typename std::enable_if_t<std::is_signed<T>::value, int> = 0>
+    inline LogStream& operator<<(const LogHex<T>&& val) noexcept;
```

Comment:
@elfenpiff `LogHex` is so small it should be passed by value instead of by reference. Objects smaller than 16 bytes are passed in registers on x86 and aarch64 which makes it faster. Will change it.

Reply:
done in #1621